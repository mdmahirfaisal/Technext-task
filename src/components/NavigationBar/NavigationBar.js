import AdbIcon from '@mui/icons-material/Adb';
import MenuIcon from '@mui/icons-material/Menu';
import AppBar from '@mui/material/AppBar';
import Box from '@mui/material/Box';
import Container from '@mui/material/Container';
import IconButton from '@mui/material/IconButton';
import Toolbar from '@mui/material/Toolbar';
import Typography from '@mui/material/Typography';
import * as React from 'react';

const NavigationBar = () => {

    return (
        <AppBar position="sticky" sx={{ bgcolor: '#fc5c65' }}>
            <Container maxWidth="xl">
                <Toolbar disableGutters>
                    <AdbIcon sx={{ display: { xs: 'none', md: 'flex' }, mr: 1 }} />
                    <Typography
                        variant="h6"
                        noWrap
                        component="a"
                        href="/"
                        sx={{
                            mr: 2,
                            display: { xs: 'none', md: 'flex' },
                            fontFamily: 'monospace',
                            fontWeight: 700,
                            letterSpacing: '.3rem',
                            color: 'inherit',
                            textDecoration: 'none',
                        }}
                    >
                        TECHNEXT
                    </Typography>

                    <Box sx={{ flexGrow: 1, display: { xs: 'flex', md: 'none' } }}>
                        <IconButton
                            size="large"
                            aria-label="account of current user"
                            aria-controls="menu-appbar"
                            aria-haspopup="true"
                            color="inherit"
                        >
                            <MenuIcon />
                        </IconButton>
                    </Box>
                    <AdbIcon sx={{ display: { xs: 'flex', md: 'none' }, mr: 1 }} />
                    <Typography
                        variant="h5"
                        noWrap
                        component="a"
                        href=""
                        sx={{
                            mr: 2,
                            display: { xs: 'flex', md: 'none' },
                            flexGrow: 1,
                            fontFamily: 'monospace',
                            fontWeight: 700,
                            letterSpacing: '.3rem',
                            color: 'inherit',
                            textDecoration: 'none',
                        }}
                    >
                        TECHNEXT
                    </Typography>
                    <p className='mx-auto text-xl mb-3 font-semibold text-gray-50'>Search data by rocket name like: falcon 1 or falcon 9</p>
                </Toolbar>

            </Container>
        </AppBar>
    );
};
export default NavigationBar;
