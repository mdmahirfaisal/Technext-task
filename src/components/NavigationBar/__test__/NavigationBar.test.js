import React from 'react'
import ReactDOM from 'react-dom'
import NavigationBar from '../NavigationBar'
import renderer from 'react-test-renderer'
import "@testing-library/jest-dom";


it('navigation bar render without crashing',
    () => {
        const div = document.createElement('div')
        ReactDOM.render(
            <NavigationBar />
            , div)
    })


it("matches navigation bar snapshot",
    () => {
        const tree = renderer.create(
            <NavigationBar />
        ).toJSON()
        expect(tree).toMatchSnapshot()
    })