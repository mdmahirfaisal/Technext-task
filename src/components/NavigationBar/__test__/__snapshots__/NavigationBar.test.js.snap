// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches navigation bar snapshot 1`] = `
<header
  className="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-c5nfbw-MuiPaper-root-MuiAppBar-root"
>
  <div
    className="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <div
      className="MuiToolbar-root MuiToolbar-regular css-1cndkqf-MuiToolbar-root"
    >
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-1uzrjc8-MuiSvgIcon-root"
        data-testid="AdbIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 16c0 3.87 3.13 7 7 7s7-3.13 7-7v-4H5v4zM16.12 4.37l2.1-2.1-.82-.83-2.3 2.31C14.16 3.28 13.12 3 12 3s-2.16.28-3.09.75L6.6 1.44l-.82.83 2.1 2.1C6.14 5.64 5 7.68 5 10v1h14v-1c0-2.32-1.14-4.36-2.88-5.63zM9 9c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm6 0c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
        />
      </svg>
      <a
        className="MuiTypography-root MuiTypography-h6 MuiTypography-noWrap css-m5x0z-MuiTypography-root"
        href="/"
      >
        TECH NEXT
      </a>
      <div
        className="MuiBox-root css-1lvtzne"
      >
        <button
          aria-controls="menu-appbar"
          aria-haspopup="true"
          aria-label="account of current user"
          className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeLarge css-1pz7awu-MuiButtonBase-root-MuiIconButton-root"
          disabled={false}
          onBlur={[Function]}
          onContextMenu={[Function]}
          onDragLeave={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex={0}
          type="button"
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
            data-testid="MenuIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
            />
          </svg>
        </button>
      </div>
      <svg
        aria-hidden={true}
        className="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-z5gn4u-MuiSvgIcon-root"
        data-testid="AdbIcon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 16c0 3.87 3.13 7 7 7s7-3.13 7-7v-4H5v4zM16.12 4.37l2.1-2.1-.82-.83-2.3 2.31C14.16 3.28 13.12 3 12 3s-2.16.28-3.09.75L6.6 1.44l-.82.83 2.1 2.1C6.14 5.64 5 7.68 5 10v1h14v-1c0-2.32-1.14-4.36-2.88-5.63zM9 9c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1zm6 0c-.55 0-1-.45-1-1s.45-1 1-1 1 .45 1 1-.45 1-1 1z"
        />
      </svg>
      <a
        className="MuiTypography-root MuiTypography-h5 MuiTypography-noWrap css-1lfs2tq-MuiTypography-root"
        href=""
      >
        TECHNEXT
      </a>
      <p
        className="mx-auto text-xl mb-3 font-semibold text-gray-50"
      >
        Search data by rocket name like: falcon 1 or falcon 9
      </p>
    </div>
  </div>
</header>
`;
